linguicator_predictor/websocket.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import websockets
+
+from linguicator_predictor.service import handle_request
+
+
+async def handle_websocket_connection(websocket, path):
+    """
+    I handle a new websocket connection by listening to requests and handling
+    the received data until the connection is closed.
+
+    :param websocket: websocket object representing the connection
+    :type  websocket: websockets.server.WebSocketServerProtocol
+
+    :param path: the connection relative path
+    :type  path: str
+
+    :returns: nothing
+    :rtype: None
+    """
+    logging.info('New connection established')
+
+    user_text = ''
+
+    # wait for user requests until the connection is closed
+    while True:
+
+        try:
+            user_text = await wait_for_user_request(websocket, user_text)
+        except websockets.exceptions.ConnectionClosedOK:
+            logging.info('Connection closed cleanly')
+            return
+        except Exception as e:
+            logging.info(f'Connection closed due unexpected error: {e}')
+            return
+
+
+async def wait_for_user_request(websocket, user_text):
+    """
+    I wait for data in the websocket connection.
+
+    :param websocket: websocket object representing the connection
+    :type  websocket: websockets.server.WebSocketServerProtocol
+
+    :param user_text: text the user typed so far
+    :type  user_text: str
+
+    :returns: the updated user text
+    :rtype: str
+    """
+    request = await websocket.recv()
+    logging.info('data received through websocket connection')
```

Comment:
We could start with uppercase here as well.

Reply:
Done